tracing/tracer.go

```diff
@@ -0,0 +1,75 @@
+package tracing
+
+import (
+	"io"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	jeagerConf "github.com/uber/jaeger-client-go/config"
+)
+
+type Tracer struct {
+	ServiceName  string
+	Provider     string
+	Logger       *logrus.Logger
+	JaegerConfig *JaegerConfig
+
+	tracer opentracing.Tracer
+	closer io.Closer
+}
+
+type JaegerConfig struct {
+	LocalAgentHostPort string
+	SamplerType        string
+	SamplerValue       float64
+	SamplerServerUrl   string
+}
+
+func (t *Tracer) Setup() {
+	switch t.Provider {
+	case "jaeger":
+		jc := jeagerConf.Configuration{
+			Sampler: &jeagerConf.SamplerConfig{
+				SamplingServerURL: t.JaegerConfig.SamplerServerUrl,
+				Type:              t.JaegerConfig.SamplerType,
+				Param:             t.JaegerConfig.SamplerValue,
+			},
+			Reporter: &jeagerConf.ReporterConfig{
+				LocalAgentHostPort: t.JaegerConfig.LocalAgentHostPort,
+			},
+		}
+
+		closer, err := jc.InitGlobalTracer(
+			t.ServiceName,
+		)
+
+		if err != nil {
+			t.Logger.Warnf("Could not initialize jaeger tracer: %s", err.Error())
+			return
+		}
+
+		t.closer = closer
+		t.tracer = opentracing.GlobalTracer()
+		t.Logger.Infof("Jaeger tracer configured!")
+	default:
```

Comment:
IMO we should have two cases here:

```
case "":
// disabled
default:
// fatal because unknown tracer
```

By explicitly failing on an unknown tracer, we make debugging much easier.

Reply:
Good point, will update.

Reply:
@aeneasr should it be a fatal error in this case if someone misconfigured their tracing config or a typo? 

The rest of Hydra can still run so I am not sure if we want to bring down everything. Maybe logging at error level is more appropriate?

Reply:
Yes it should, because it's an obvious mistake so hydra should make it explicit that it's misconfigured. Better to have a service not start than to discover that some feature didn't work all along that you now urgently need to debug something.